queue/group_remote_mongo.go

```diff
@@ -231,33 +173,34 @@ func (g *remoteMongoQueueGroup) Queues(ctx context.Context) []string {
 	g.mu.RLock()
 	defer g.mu.RUnlock()
 
-	out, _ := g.getExistingCollections(ctx, g.client, g.dbOpts.DB, g.opts.Prefix) // nolint
+	out, _ := g.getExistingCollections(ctx, g.opts.Queue.DB.Client, g.opts.Queue.DB.DB, g.opts.Queue.DB.Collection) // nolint
 
 	return out
 }
 
-func (g *remoteMongoQueueGroup) startProcessingRemoteQueue(ctx context.Context, coll string) (amboy.Queue, error) {
+func (g *remoteMongoQueueGroup) startProcessingRemoteQueue(ctx context.Context, coll string, opts ...amboy.QueueOptions) (amboy.Queue, error) {
 	coll = trimJobsSuffix(coll)
-	q, err := g.opts.constructor(ctx, coll)
+	mdbOpts, err := getMongoDBQueueOptions(opts...)
 	if err != nil {
-		return nil, errors.Wrap(err, "constructing queue")
+		return nil, errors.Wrap(err, "getting queue options")
 	}
 
-	var d remoteQueueDriver
-	if g.client != nil {
-		d, err = openNewMongoDriver(ctx, coll, g.dbOpts, g.client)
-		if err != nil {
-			return nil, errors.Wrap(err, "creating and opening driver")
-		}
-	} else {
-		d, err = newMongoDriver(coll, g.dbOpts)
-		if err != nil {
-			return nil, errors.Wrap(err, "creating and opening driver")
-		}
-	}
-	if err := q.SetDriver(d); err != nil {
-		return nil, errors.Wrap(err, "setting driver")
+	queueOpts := mergeMongoDBQueueOptions(append([]MongoDBQueueOptions{g.opts.Queue}, mdbOpts...)...)
```

Comment:
Should passed in opts take precedence?

Reply:
Same thing as above - later options take precedence over earlier ones.